mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1609,6 +1610,20 @@ static LogicalResult reductionPreconditions(LinalgOp op) {
 }
 
 static LogicalResult vectorizeDynamicLinalgOpPrecondition(linalg::LinalgOp op) {
+  if (auto conv = dyn_cast<linalg::DepthwiseConv1DNwcWcOp>(op.getOperation())) {
```

Comment:
could we move all this conv specific logic to a specific `*Precondition` function?